pkg/remote/codec/header_codec_test.go

```diff
@@ -88,6 +91,76 @@ func TestTTHeaderCodecWithTransInfo(t *testing.T) {
 	test.Assert(t, flag == uint16(HeaderFlagSupportOutOfOrder))
 }
 
+func TestTTHeaderCodecWithTransInfoWithGDPRToken(t *testing.T) {
+	ctx := context.Background()
+	intKVInfo := prepareIntKVInfo()
+	strKVInfo := prepareStrKVInfoWithGDPRToken()
+	sendMsg := initClientSendMsg(transport.TTHeader)
+	sendMsg.TransInfo().PutTransIntInfo(intKVInfo)
+	sendMsg.TransInfo().PutTransStrInfo(strKVInfo)
+	sendMsg.Tags()[HeaderFlagsKey] = HeaderFlagSupportOutOfOrder
+
+	// encode
+	out := remote.NewWriterBuffer(256)
+	totalLenField, err := ttHeaderCodec.encode(ctx, sendMsg, out)
+	binary.BigEndian.PutUint32(totalLenField, uint32(out.MallocLen()-Size32+mockPayloadLen))
+	test.Assert(t, err == nil, err)
+
+	// decode
+	recvMsg := initServerRecvMsg()
+	buf, err := out.Bytes()
+	test.Assert(t, err == nil, err)
+	test.Assert(t, strings.Contains(string(buf), strKVInfo[transmeta.GDPRToken]), string(buf))
+	in := remote.NewReaderBuffer(buf)
+	err = ttHeaderCodec.decode(ctx, recvMsg, in)
+	test.Assert(t, err == nil, err)
+	test.Assert(t, recvMsg.PayloadLen() == mockPayloadLen, recvMsg.PayloadLen())
+
+	intKVInfoRecv := recvMsg.TransInfo().TransIntInfo()
+	strKVInfoRecv := recvMsg.TransInfo().TransStrInfo()
+	test.DeepEqual(t, intKVInfoRecv, intKVInfo)
+	test.DeepEqual(t, strKVInfoRecv, map[string]string{transmeta.HeaderTransRemoteAddr: "mockRemoteAddr"})
+	flag := recvMsg.Tags()[HeaderFlagsKey]
+	test.Assert(t, flag != nil)
+	test.Assert(t, flag == uint16(HeaderFlagSupportOutOfOrder))
+}
+
+func TestTTHeaderCodecWithTransInfoFromMetaInfoGDPRToken(t *testing.T) {
+	ctx := context.Background()
+	intKVInfo := prepareIntKVInfo()
+	ctx = metainfo.WithValue(ctx, "gdpr-token", "test token")
+	token, ok := metainfo.GetValue(ctx, "gdpr-token")
+	test.Assert(t, ok == true)
+	sendMsg := initClientSendMsg(transport.TTHeader)
+	sendMsg.TransInfo().PutTransIntInfo(intKVInfo)
+	sendMsg.TransInfo().PutTransStrInfo(map[string]string{transmeta.GDPRToken: token})
```

Comment:
should not put the token directly here,  you can call MetainfoClientHandler.WriteMeta to do it